test/celeritas/neutron/NeutronElastic.test.cc

```diff
@@ -0,0 +1,310 @@
+    {
+        this->set_inc_particle(pdg::neutron(), MevEnergy{energy});
+        ChipsNeutronElasticInteractor interact(
+            shared, this->particle_track(), this->direction(), isotope_he4);
+        Interaction result = interact(rng_engine);
+
+        // Check scattered energy and angle at each incident neutron energy
+        EXPECT_SOFT_EQ(result.energy.value(), expected_energy[i]);
+        EXPECT_SOFT_EQ(dot_product(result.direction, this->direction()),
+                       expected_angle[i]);
+        energy *= factor;
+    }
+}
+
+TEST_F(NeutronElasticTest, stress_test)
+{
+    // Set the target isotope : Cu65
+    ElementComponentId el_id{1};
+    IsotopeComponentId iso_id{1};
+    IsotopeView const isotope = this->material_track()
+                                    .make_material_view()
+                                    .make_element_view(el_id)
+                                    .make_isotope_view(iso_id);
+
+    // Sample interaction
+    NeutronElasticRef shared = model_->host_ref();
+    RandomEngine& rng_engine = this->rng();
+
+    // Produce samples from the incident energy
+    std::vector<double> neutron_energy;
+    double const expected_energy[] = {9.6937900401599116e-05,
+                                      0.0096967590954386649,
+                                      0.96066635424842617,
+                                      99.915949160022208,
+                                      9999.9519447032781};
+    std::vector<double> cos_theta;
+    double const expected_angle[] = {-0.0062993051693808555,
+                                     0.0036712145032219813,
+                                     -0.29680452456419526,
+                                     0.97426025128623828,
+                                     0.9999755334707946};
+
+    int const num_sample = 100;
+    double const weight = 1.0 / static_cast<double>(num_sample);
+    std::vector<double> inc_energy = {1e-4, 0.01, 1., 100., 10000.};
+    std::vector<int> avg_rng_samples;
+    for (auto i : range(inc_energy.size()))
+    {
+        this->set_inc_particle(pdg::neutron(), MevEnergy{inc_energy[i]});
+        ChipsNeutronElasticInteractor interact(
+            shared, this->particle_track(), this->direction(), isotope);
+
+        double sum_energy{0.};
+        double sum_angle{0.};
+        int sum_count{0};
+        for ([[maybe_unused]] int j : range(num_sample))
+        {
+            // Sample scattered neutron energy and angle
+            Interaction result = interact(rng_engine);
+            sum_energy += result.energy.value();
+            sum_angle += result.direction[2];
+            sum_count += rng_engine.count();
+        }
+        avg_rng_samples.push_back(sum_count / num_sample);
+        EXPECT_SOFT_EQ(weight * sum_energy, expected_energy[i]);
+        EXPECT_SOFT_EQ(weight * sum_angle, expected_angle[i]);
+    }
+    static int const expected_avg_rng_samples[] = {202, 602, 1103, 1703, 2303};
```

Comment:
That's a lot of RNG samples for a single collision... and I don't even see a rejection loop in the interactor. Am I missing something?

Reply:
Okay, I added reset for each sampling and the average rng counts are 
expected_avg_rng_samples[] = {4, 4, 6, 6, 6};
which seems reasonable.